tests/unit/grains/core_test.py

```diff
@@ -28,6 +28,15 @@
 
 # Globals
 core.__salt__ = {}
+core.__opts__ = {}
```

Comment:
Here `__role` is missing, as the default config was _never_ applied to the `__opts__`. Whilst I am not sure why the heck Ubuntu is checking it and RH is not, I'd apply the default config here or at least fake some of those values.